src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -56,10 +57,34 @@ public final class QueryInsightsListener extends SearchRequestOperationsListener
      */
     @Inject
     public QueryInsightsListener(final ClusterService clusterService, final QueryInsightsService queryInsightsService) {
+        super(false); // Disable query insights listener and service initially
         this.clusterService = clusterService;
         this.queryInsightsService = queryInsightsService;
-        // Setting endpoints set up for top n queries, including enabling top n queries, window size and top n size
-        // Expected metricTypes are Latency, CPU and Memory.
+        initialize();
```

Comment:
Can be rewritten as `this(clusterService, queryInsightsService, false);`

Reply:
Call to this() has to be the first call in the method body but we already called super(). Don't think we can call both super() and this() in the same constructor.